sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/ShardCheckpoint.java

```diff
@@ -155,6 +180,23 @@ private boolean checkpointIsInTheMiddleOfAUserRecord() {
     return shardIteratorType == AFTER_SEQUENCE_NUMBER && subSequenceNumber != null;
   }
 
+  public CompletableFuture<Void> subscribeToShard(
+      boolean resubscribe,
+      SimplifiedKinesisClient kinesisClient,
+      final SubscribeToShardResponseHandler.Visitor visitor,
+      final Consumer<Throwable> onError) {
+    if (resubscribe) {
+      return kinesisClient.subscribeToShard(
```

Comment:
Don't we have to use the `continuationSequenceNumber` to resubscribe? E.g. if we want to resubscribe to a large stream from the beginning (trim horizon) we're likely not able to process everything within one timeout interval.